app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ActiveRecord::Base
+  has_many :dogs
+
+  def self.find_or_create_from_auth(auth)
```

Comment:
Assignment Branch Condition size for find_or_create_from_auth is too high. [19.03/15]